dev/src/reference.ts

```diff
@@ -843,6 +843,7 @@ class FieldFilterInternal extends FilterInternal {
       case 'GREATER_THAN_OR_EQUAL':
       case 'LESS_THAN':
       case 'LESS_THAN_OR_EQUAL':
+      case 'NOT_EQUAL':
```

Comment:
wonder why we didn't include "!=" in inequality before.

Reply:
in the main branch, `col.where(a != 5).startAt(7);` will be validated with error message: `Error: Too many cursor values specified. The specified values must match the orderBy() constraints of the query.`

if we do it with document reference, which bypasses the validation, BE will return error: `order by clause cannot contain more fields after the key`. While if it is not "!=" but other inequality operators, BE will return empty result.

Reply:
If I understand correctly, prior to adding `case 'NOT_EQUAL'`, the user would _always_ get an error when using cursors (either from sdk or server). This fixes the issue.

Reply:
yes, unless the user was SUPER cautious to always add `orderBy`, which makes the query no longer relies on `createImplicitOrderBy` and `isInequalityFilter()` .  Like this:
```
col.where(a != 5).orderby('a').startAt(7);
col.where(a != 5).orderby('a').startAt(docSnap);
col.where(a != 5).orderby('a').startAt(docRef); / /works but wrong value returned

```